lib/db-item.ts

```diff
@@ -1,50 +1,65 @@
-import {getClient} from './db-client';
+import oracledb from 'oracledb';
+import { getClient } from './db-client';
+import { COLLECTION } from './private/enums';
 
 /**
  * The interface for a database item
  */
-export interface IDbItem extends Record<string, any> {
+export interface IDbItem {
   readonly id: string;
+  readonly collectionName: string;
+  readonly existsInDb: boolean;
 }
 
 /**
- * The attributes required to query a database item
+ * A database item. All database items should extend this class.
  */
-export interface DbItemAttributes {
+export abstract class DbItem implements IDbItem {
   /**
-   * The id of the database item
+   * The id of the item in a collection
    */
-  readonly id: string;
+  public readonly id: string;
 
   /**
-   * The collection the item lives in
+   * The collection this database item belongs to
    */
-  readonly collectionName: string;
-}
+  public readonly collectionName: COLLECTION;
 
-/**
- * A database item. All database items should extend this class.
- */
-export class DbItem implements IDbItem {
+  /**
+   * Check if the object exists within the database already
+   *
+   * Useful for determining whether or not to insert or replace an item.
+   */
+  get existsInDb(): boolean {
+    return this.#existsInDb;
+  }
 
-  public static async findDbItemFromAttributes({id, collectionName}: DbItemAttributes): Promise<IDbItem | null> {
-    const client = await getClient();
-    const content = (await client.findDocument(collectionName, id))?.getContent();
-    return content as IDbItem ?? null;
+  #existsInDb: boolean;
```

Comment:
Actually makes the property private!

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields